src/EEG_1SCU_CM.py

```diff
@@ -1,177 +1,182 @@
+        train_loss: Loss metric for training data.
+        test_step_outputs (List[Dict[str, torch.Tensor]]): A list to store test step outputs,
+            including test loss and accuracy.
+
+    Methods:
+        forward(x: torch.Tensor) -> torch.Tensor: Performs forward pass through the model.
+        training_step(batch: Tuple[torch.Tensor, torch.Tensor], batch_idx: int) -> torch.Tensor:
+            Defines the training step logic.
+        test_step(batch: Tuple[torch.Tensor, torch.Tensor], batch_idx: int) -> None:
+            Defines the test step logic.
+        on_test_epoch_end() -> None:
+            Calculates and logs test loss and accuracy at the end of each testing epoch.
+        configure_optimizers() -> torch.optim.Optimizer:
+            Configures the optimizer used for training the model.
+    """
+
+    def __init__(self, config: Dict[str, Any]) -> None:
+        """
+        Initializes the SCU_Model with the provided configuration.
+
+        Args:
+            config (Dict[str, Any]): A dictionary containing configuration parameters for the SCU model.
+        """
+        super().__init__()
+        self.config = config
+        self.model = SCU(self.config).double()
+        self.criterion = nn.CrossEntropyLoss()
+        self.test_acc = Accuracy(num_classes=config["num_class"], task="multiclass")
+        self.test_step_outputs = []
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """
+        Performs a forward pass through the SCU model.
+
+        Args:
+            x (torch.Tensor): Input data tensor.
+
+        Returns:
+            torch.Tensor: Output tensor from the SCU model.
+        """
+        return self.model(x)
+
+    def training_step(
+        self, batch: Tuple[torch.Tensor, torch.Tensor], batch_idx: int
+    ) -> torch.Tensor:
+        """
+        Defines the training step logic.
+
+        Args:
+            batch (Tuple[torch.Tensor, torch.Tensor]): A tuple containing input data and corresponding labels.
+            batch_idx (int): Index of the batch.
+
+        Returns:
+            torch.Tensor: Training loss tensor.
+        """
+        x, y = batch
+        y_hat = self.model(x)
+        loss = self.criterion(y_hat, y.long())
+        self.log("train_loss", loss, prog_bar=True)
+        return loss
+
+    def test_step(self, batch: Tuple[torch.Tensor, torch.Tensor], batch_idx: int):
+        """
+        Defines the test step logic.
+
+        Args:
+            batch (Tuple[torch.Tensor, torch.Tensor]): A tuple containing input data and corresponding labels.
+            batch_idx (int): Index of the batch.
+        """
+        x, y = batch
+        y_hat = self.model(x)
+        test_loss = self.criterion(y_hat, y.long())
+        test_acc = self.test_acc(torch.argmax(y_hat, dim=1), y.long())
+        self.test_step_outputs.append(
+            {"test_loss": test_loss, "test_acc": test_acc, "y": y, "y_hat": y_hat}
+        )
```

Comment:
Might be worth revising this as I think this the test_step should return something?